index.js

```diff
@@ -1,8 +1,24 @@
 'use strict';
-module.exports = function (str) {
-	if (typeof str !== 'string') {
-		throw new TypeError('Expected a string');
+
+var replaceBuf = function (buf) {
+	var crlf = '\r\n';
+	var i = buf.indexOf(crlf);
+
+	if (i === -1) {
+		return buf;
+	}
+
+	var start = buf.slice(0, i);
+	var end = replaceBuf(buf.slice(i + crlf.length));
+	var len = i + start.length + end.length;
+
+	return Buffer.concat([start, '\n', end], len);
+};
+
+module.exports = function (x) {
+	if (typeof x !== 'string' && !Buffer.isBuffer(x)) {
+		throw new TypeError('Expected a string or a buffer');
```

Comment:
Unrelated, but I've been thinking about doing:

```js
throw new TypeError('Expected `string` or `Buffer`, got `' + x + '` (' + (typeof x) + ')');
```

Result:

```
TypeError: Expected `string` or `Buffer`, got `foo` (string)
```

That way the user can see the actual value too. Also dropped the `a`'s. What do you guys think?